### ISSUE TITLE
Clean up importScripts / require? interface

### ISSUE THREAD

**Issue by gzm0** [scala-js/scala-js#457]
The sbt-plugin of Scala.js currently provides an ad-hoc way to import JS files/libraries on the classpath through the `importScripts` method [specified as in web workers](https://developer.mozilla.org/en-US/docs/Web/Guide/Performance/Using_web_workers#Importing_scripts_and_libraries). This basically boils down to: "Put the code of the scripts where I write `importScripts`.

This is not the only way how multiple JS files can be handled: There is [CommonJS's module format](http://www.commonjs.org/specs/modules/1.0/) which is supported by Node.js with some extensions out of spec: You may append ".js" to the module name and the loaded module may share the global scope with the calling context (configured via environment var). Browser support via aggregation tools ([Webmake](https://github.com/medikoo/modules-webmake) / [Browserify](http://browserify.org/))

Further, there is [RequireJS](http://requirejs.org/) which has its own module system but it works in the browser and on node.js.

Long story short, there are very different ways of doing things and they are not compatible on how the module source must be written. How can we expose/support these things in the sbt-runner / tester such that it can be easily integrated in potentially larger develop/test setup?
